src/test/java/com/xceptance/neodymium/tests/NeodymiumTestSelfTest.java

```diff
@@ -0,0 +1,177 @@
+            }
+        });
+        checkFail(result, 2, 0, 1, errorMessage);
+    }
+
+    @Test
+    public void testCheckFailedTwoFromTwoNumber() throws NoSuchFieldException, SecurityException, IllegalArgumentException, IllegalAccessException
+    {
+        Result result = createResult(2, 0, new HashMap<String, Throwable>()
+        {
+            private static final long serialVersionUID = 3L;
+
+            {
+                this.put(name.getMethodName() + "1", new RuntimeException("This is RuntimeException 1"));
+                this.put(name.getMethodName() + "2", new RuntimeException("This is RuntimeException 1"));
+            }
+        });
+
+        // no assertion for failure message, as one is only possible for single fail
+        checkFail(result, 2, 0, 2);
+    }
+
+    @Test
+    public void testCheckFailedTwoFromTwoOneFailureMessage() throws NoSuchFieldException, SecurityException, IllegalArgumentException, IllegalAccessException
+    {
+        Result result = createResult(2, 0, new HashMap<String, Throwable>()
+        {
+            private static final long serialVersionUID = 4L;
+
+            {
+                this.put(name.getMethodName() + "1", new RuntimeException("This is RuntimeException 1"));
+                this.put(name.getMethodName() + "2", new RuntimeException("This is RuntimeException 1"));
+            }
+        });
+
+        // no assertion for failure message, as one is only possible for single fail
+        checkFail(result, 2, 0, 2, "This is RuntimeException 1");
+    }
+
+    @Test
+    public void testCheckFailedTwoFromTwoTwoFailureMessages() throws NoSuchFieldException, SecurityException, IllegalArgumentException, IllegalAccessException
+    {
+        Result result = createResult(2, 0, new HashMap<String, Throwable>()
+        {
+            private static final long serialVersionUID = 5L;
+
+            {
+                this.put(name.getMethodName() + "1", new RuntimeException("This is RuntimeException 1"));
+                this.put(name.getMethodName() + "2", new RuntimeException("This is RuntimeException 2"));
+            }
+        });
+        HashMap<String, String> expectedFailureMessages = new HashMap<String, String>()
+        {
+            private static final long serialVersionUID = 6L;
+
+            {
+                this.put(name.getMethodName() + "1", "This is RuntimeException 1");
+                this.put(name.getMethodName() + "2", "This is RuntimeException 2");
+            }
+        };
+        // no assertion for failure message, as one is only possible for single fail
+        checkFail(result, 2, 0, 2, expectedFailureMessages);
+    }
+
+    @Test
+    public void testCheckOneFailedOneIgnoredFromTwo() throws NoSuchFieldException, SecurityException, IllegalArgumentException, IllegalAccessException
+    {
+        final String errorMessage = "This is RuntimeException 1";
+        Result result = createResult(2, 1, new HashMap<String, Throwable>()
```

Comment:
See comment https://github.com/Xceptance/neodymium-library/pull/137/files#r479258825

Reply:
Done

Reply:
Building the failures map can be simplified (similar to `testCheckFailedOneFromOne`)

Reply:
done